spec/request/salesman/update_spec.rb

```diff
@@ -1,194 +1,161 @@
 require 'rails_helper'
+require 'faker'
 
 describe '::Api::V1::SalesmanController', type: :request do
+
+  before do
+    I18n.default_locale = :en
+    execute_actions
+  end
+
+  let(:url) { '/api/v1/salesman' }
+
+  let(:execute_actions) {}
+
+  let(:body) { JSON.parse response.body }
+
+  let(:valid_params_create) do
+    {
+      name: Faker::Name.name,
+      status: Faker::Name.name,
+      phone: [
+        { number: Faker::PhoneNumber.cell_phone, whatsapp: true },
+        { number: Faker::PhoneNumber.cell_phone, whatsapp: true }
+      ]
+    }
+  end
+
+  let(:valid_params_update) do
+    {
+      name: Faker::Name.name,
+      status: Faker::Name.name
+    }
+  end
+
+  let(:valid_params_phone) do
+    {
+      phone: [
+        { number: Faker::PhoneNumber.cell_phone, whatsapp: true }
+      ]
+    }
+  end
+
   describe 'When need update a salesman' do
+    let(:execute_actions) do
+      post url, params: { salesman: valid_params_create }
+    end
     context 'when request attributes are valid' do
       it 'Should return status code 200' do
-        params = {
-          name: 'joao updated',
-          status: 'ok',
-          phone: [
-            { number: '8288776501', whatsapp: true }
-          ]
-        }
-        post '/api/v1/salesman', params: { salesman: params }
-
-        body = JSON.parse response.body
-
-        put "/api/v1/salesman/#{body['data']['id']}", params: { salesman: params }
+
+        put "#{url}/#{body['data']['id']}", params: { salesman: valid_params_update }
```

Comment:
I suggest to extract this execute action for a before or update the `let(:execute_actions)` for this context.